corehq/apps/reports/standard/cases/case_data.py

```diff
@@ -555,6 +559,209 @@ def close_case_view(request, domain, case_id):
     return HttpResponseRedirect(reverse('case_data', args=[domain, case_id]))
 
 
+MAX_CASE_COUNT = 10
+MAX_SUBCASE_DEPTH = 3
+
+
+@location_safe
+def get_cases_and_forms_for_deletion(request, domain, case_id):
+    delete_cases = []  # list of cases to be soft deleted
+    delete_forms = []  # list of forms to be soft deleted
+
+    # For formatting the list of cases/submission forms
+    cases = {}  # structured like: {case: {form: {case touched by form: actions taken by form}}}
+    case_names = {}  # {case_id: case_name}
+    form_names = {}  # {form_ids: form name}
+    reopened_cases = {}
+    affected_cases = {}
+
+    update_actions = [const.CASE_ACTION_INDEX,
+                      const.CASE_ACTION_UPDATE,
+                      const.CASE_ACTION_ATTACHMENT,
+                      const.CASE_ACTION_COMMTRACK,
+                      const.CASE_ACTION_REBUILD]
+
+    def get_forms_for_deletion_from_case(case, subcase_count):
+        delete_cases.append(case.case_id)
+        if len(delete_cases) > MAX_CASE_COUNT or subcase_count >= MAX_SUBCASE_DEPTH:
+            raise ValueError("Too many cases to delete")
```

Comment:
`ValueError` is a generic error that can be raised by other Python operations. It would be safer to create a custom error type to be used for this purpose. It could be as simple as
```py
class TooManyCases(ValueError):
    pass
```